src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -519,6 +530,154 @@ def _hypercube(
 NdArrayField.register_builtin_func(fbuiltins.where, _make_builtin("where", "where"))
 
 
+def _compute_mask_ranges(
+    mask: core_defs.NDArrayObject,
+) -> list[tuple[bool, common.UnitRange]]:
+    """Take a 1-dimensional mask and return a sequence of mappings from boolean values to ranges."""
+    # TODO: does it make sense to upgrade this naive algorithm to numpy?
```

Comment:
if used in k, probably not relevant, if used in the horizontal, probably this is a performance problem